NuGet/New-BcNuGetPackage.ps1

```diff
@@ -129,51 +118,45 @@ Function New-BcNuGetPackage {
+            $XmlObjectWriter.WriteAttributeString("url", $githubRepository)
             $XmlObjectWriter.WriteEndElement()
         }
-        $XmlObjectWriter.WriteEndElement()
-        $XmlObjectWriter.WriteStartElement("files")
-        $appFiles | ForEach-Object {
-            $XmlObjectWriter.WriteStartElement("file")
-            $appFileName = [System.IO.Path]::GetFileName($_)
-            $XmlObjectWriter.WriteAttributeString("src", $appFileName );
-            $XmlObjectWriter.WriteAttributeString("target", $appFileName);
+        $XmlObjectWriter.WriteStartElement("dependencies")
+        $appJson.dependencies | ForEach-Object {
+            $id = $dependencyIdTemplate.replace('{id}',$_.id).replace('{name}',[nuGetFeed]::Normalize($_.name)).replace('{publisher}',[nuGetFeed]::Normalize($_.publisher))
+            $XmlObjectWriter.WriteStartElement("dependency")
+            $XmlObjectWriter.WriteAttributeString("id", $id)
+            $XmlObjectWriter.WriteAttributeString("version", $_.Version)
             $XmlObjectWriter.WriteEndElement()
         }
-        $testAppFiles | ForEach-Object {
-            $XmlObjectWriter.WriteStartElement("file")
-            $appFileName = [System.IO.Path]::GetFileName($_)
-            $XmlObjectWriter.WriteAttributeString("src", "$testsFolderName\$appFileName" );
-            $XmlObjectWriter.WriteAttributeString("target", "$testsFolderName\$appFileName" );
+        if ($appJson.PSObject.Properties.Name -eq 'Application' -and $appJson.Application) {
+            $XmlObjectWriter.WriteStartElement("dependency")
+            $XmlObjectWriter.WriteAttributeString("id", $applicationDependencyId)
+            $XmlObjectWriter.WriteAttributeString("version", $appJson.Application)
             $XmlObjectWriter.WriteEndElement()
         }
-        $dependencyAppFiles | ForEach-Object {
-            $XmlObjectWriter.WriteStartElement("file")
-            $appFileName = [System.IO.Path]::GetFileName($_)
-            $XmlObjectWriter.WriteAttributeString("src", "$dependenciesFolderName\$appFileName" );
-            $XmlObjectWriter.WriteAttributeString("target", "$dependenciesFolderName\$appFileName" );
+        if ($appJson.PSObject.Properties.Name -eq 'Platform' -and  $appJson.Platform) {
+            $XmlObjectWriter.WriteStartElement("dependency")
+            $XmlObjectWriter.WriteAttributeString("id", $platformDependencyId)
+            $XmlObjectWriter.WriteAttributeString("version", $appJson.Platform)
             $XmlObjectWriter.WriteEndElement()
         }
         $XmlObjectWriter.WriteEndElement()
         $XmlObjectWriter.WriteEndElement()
+        $XmlObjectWriter.WriteStartElement("files")
+        $XmlObjectWriter.WriteStartElement("file")
+        $appFileName = [System.IO.Path]::GetFileName($appfile)
+        $XmlObjectWriter.WriteAttributeString("src", $appFileName );
+        $XmlObjectWriter.WriteAttributeString("target", $appFileName);
+        $XmlObjectWriter.WriteEndElement()

```

Comment:
Consider either requiring a .nuspec file as input or construcitng a nuspec and using the nuget.exe to create the package from the nuspec.

Reply:
A general thing in ContainerHelper is to have as few dependencies as possible on external tools that needs to be installed.